test/memoized.spec.ts

```diff
@@ -0,0 +1,25 @@
+// Import Third-party Dependencies
+import { expect } from "chai";
+
+// Import Internal Dependencies
+import { RC } from "../src/rc.js";
+import { memoized } from "../src/index.js";
+
+describe("memoized", () => {
+  let memoizedValue: Partial<RC> | null = null;
+  afterEach(() => {
+    memoizedValue = null;
```

Comment:
We probably need a third method not exported to reset to null the memoizedValue.

Reply:
Hi @fraxken, I can't properly visualize a use case for the `clearMemoized` method, I made the use case below which should allow me to reset the value of `memoizedValue` whenever the `efaultValue` is to null.
``` 
export function memoized(options: IMemoizedOptions): Partial<RC> | null {
  const { defaultValue = null } = options;

  if (defaultValue === null) {
    clearMemoized();
  }

  return memoizedValue ?? defaultValue;
}

function clearMemoized(): void {
  memoizedValue = null;
}

```